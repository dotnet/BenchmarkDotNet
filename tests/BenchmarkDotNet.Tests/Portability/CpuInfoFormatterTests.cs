#if CLASSIC
using System;
using System.Text;
using ApprovalTests;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using BenchmarkDotNet.Helpers;
using Xunit;

namespace BenchmarkDotNet.Tests.Portability
{
    [UseReporter(typeof(DiffReporter))]
    [UseApprovalSubdirectory("ApprovedFiles")]
    public class CpuInfoFormatterTests
    {
        [Fact]
        public void BuildProcessorInfoCaptionTest()
        {
            var captions = new StringBuilder();
            foreach (var processorName in new[] { null, "", "Intel" })
            foreach (var physicalProcessorCount in new int?[] { null, 0, 1, 2 })
            foreach (var physicalCoreCount in new int?[] { null, 0, 1, 2 })
            foreach (var logicalCoreCount in new int?[] { null, 0, 1, 2 })
                captions.AppendLine(CpuInfoFormatter.Format(new MockCpuInfo(processorName, physicalProcessorCount, physicalCoreCount, logicalCoreCount)));
            
            Approvals.Verify(captions.ToString());
        }

        private class MockCpuInfo : ICpuInfo
        {
            public MockCpuInfo(string processorName, int? physicalProcessorCount, int? physicalCoreCount, int? logicalCoreCount)
            {
                PhysicalCoreCount = physicalCoreCount;
                PhysicalProcessorCount = physicalProcessorCount;
                LogicalCoreCount = logicalCoreCount;
                ProcessorName = processorName;
            }

            public int? PhysicalCoreCount { get; }
            public int? PhysicalProcessorCount { get; }
            public int? LogicalCoreCount { get; }
            public string ProcessorName { get; }
        }
    }
}
#endif