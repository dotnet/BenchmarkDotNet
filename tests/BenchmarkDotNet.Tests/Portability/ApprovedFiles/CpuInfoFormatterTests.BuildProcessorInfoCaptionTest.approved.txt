Unknown processor
Unknown processor
Unknown processor, 1 logical core
Unknown processor, 2 logical cores
Unknown processor
Unknown processor
Unknown processor, 1 logical core
Unknown processor, 2 logical cores
Unknown processor, 1 physical core
Unknown processor, 1 physical core
Unknown processor, 1 logical core and 1 physical core
Unknown processor, 2 logical cores and 1 physical core
Unknown processor, 2 physical cores
Unknown processor, 2 physical cores
Unknown processor, 1 logical core and 2 physical cores
Unknown processor, 2 logical cores and 2 physical cores
Unknown processor
Unknown processor
Unknown processor, 1 logical core
Unknown processor, 2 logical cores
Unknown processor
Unknown processor
Unknown processor, 1 logical core
Unknown processor, 2 logical cores
Unknown processor, 1 physical core
Unknown processor, 1 physical core
Unknown processor, 1 logical core and 1 physical core
Unknown processor, 2 logical cores and 1 physical core
Unknown processor, 2 physical cores
Unknown processor, 2 physical cores
Unknown processor, 1 logical core and 2 physical cores
Unknown processor, 2 logical cores and 2 physical cores
Unknown processor, 1 CPU
Unknown processor, 1 CPU
Unknown processor, 1 CPU, 1 logical core
Unknown processor, 1 CPU, 2 logical cores
Unknown processor, 1 CPU
Unknown processor, 1 CPU
Unknown processor, 1 CPU, 1 logical core
Unknown processor, 1 CPU, 2 logical cores
Unknown processor, 1 CPU, 1 physical core
Unknown processor, 1 CPU, 1 physical core
Unknown processor, 1 CPU, 1 logical core and 1 physical core
Unknown processor, 1 CPU, 2 logical cores and 1 physical core
Unknown processor, 1 CPU, 2 physical cores
Unknown processor, 1 CPU, 2 physical cores
Unknown processor, 1 CPU, 1 logical core and 2 physical cores
Unknown processor, 1 CPU, 2 logical cores and 2 physical cores
Unknown processor, 2 CPU
Unknown processor, 2 CPU
Unknown processor, 2 CPU, 1 logical core
Unknown processor, 2 CPU, 2 logical cores
Unknown processor, 2 CPU
Unknown processor, 2 CPU
Unknown processor, 2 CPU, 1 logical core
Unknown processor, 2 CPU, 2 logical cores
Unknown processor, 2 CPU, 1 physical core
Unknown processor, 2 CPU, 1 physical core
Unknown processor, 2 CPU, 1 logical core and 1 physical core
Unknown processor, 2 CPU, 2 logical cores and 1 physical core
Unknown processor, 2 CPU, 2 physical cores
Unknown processor, 2 CPU, 2 physical cores
Unknown processor, 2 CPU, 1 logical core and 2 physical cores
Unknown processor, 2 CPU, 2 logical cores and 2 physical cores
Unknown processor
Unknown processor
Unknown processor, 1 logical core
Unknown processor, 2 logical cores
Unknown processor
Unknown processor
Unknown processor, 1 logical core
Unknown processor, 2 logical cores
Unknown processor, 1 physical core
Unknown processor, 1 physical core
Unknown processor, 1 logical core and 1 physical core
Unknown processor, 2 logical cores and 1 physical core
Unknown processor, 2 physical cores
Unknown processor, 2 physical cores
Unknown processor, 1 logical core and 2 physical cores
Unknown processor, 2 logical cores and 2 physical cores
Unknown processor
Unknown processor
Unknown processor, 1 logical core
Unknown processor, 2 logical cores
Unknown processor
Unknown processor
Unknown processor, 1 logical core
Unknown processor, 2 logical cores
Unknown processor, 1 physical core
Unknown processor, 1 physical core
Unknown processor, 1 logical core and 1 physical core
Unknown processor, 2 logical cores and 1 physical core
Unknown processor, 2 physical cores
Unknown processor, 2 physical cores
Unknown processor, 1 logical core and 2 physical cores
Unknown processor, 2 logical cores and 2 physical cores
Unknown processor, 1 CPU
Unknown processor, 1 CPU
Unknown processor, 1 CPU, 1 logical core
Unknown processor, 1 CPU, 2 logical cores
Unknown processor, 1 CPU
Unknown processor, 1 CPU
Unknown processor, 1 CPU, 1 logical core
Unknown processor, 1 CPU, 2 logical cores
Unknown processor, 1 CPU, 1 physical core
Unknown processor, 1 CPU, 1 physical core
Unknown processor, 1 CPU, 1 logical core and 1 physical core
Unknown processor, 1 CPU, 2 logical cores and 1 physical core
Unknown processor, 1 CPU, 2 physical cores
Unknown processor, 1 CPU, 2 physical cores
Unknown processor, 1 CPU, 1 logical core and 2 physical cores
Unknown processor, 1 CPU, 2 logical cores and 2 physical cores
Unknown processor, 2 CPU
Unknown processor, 2 CPU
Unknown processor, 2 CPU, 1 logical core
Unknown processor, 2 CPU, 2 logical cores
Unknown processor, 2 CPU
Unknown processor, 2 CPU
Unknown processor, 2 CPU, 1 logical core
Unknown processor, 2 CPU, 2 logical cores
Unknown processor, 2 CPU, 1 physical core
Unknown processor, 2 CPU, 1 physical core
Unknown processor, 2 CPU, 1 logical core and 1 physical core
Unknown processor, 2 CPU, 2 logical cores and 1 physical core
Unknown processor, 2 CPU, 2 physical cores
Unknown processor, 2 CPU, 2 physical cores
Unknown processor, 2 CPU, 1 logical core and 2 physical cores
Unknown processor, 2 CPU, 2 logical cores and 2 physical cores
Intel
Intel
Intel, 1 logical core
Intel, 2 logical cores
Intel
Intel
Intel, 1 logical core
Intel, 2 logical cores
Intel, 1 physical core
Intel, 1 physical core
Intel, 1 logical core and 1 physical core
Intel, 2 logical cores and 1 physical core
Intel, 2 physical cores
Intel, 2 physical cores
Intel, 1 logical core and 2 physical cores
Intel, 2 logical cores and 2 physical cores
Intel
Intel
Intel, 1 logical core
Intel, 2 logical cores
Intel
Intel
Intel, 1 logical core
Intel, 2 logical cores
Intel, 1 physical core
Intel, 1 physical core
Intel, 1 logical core and 1 physical core
Intel, 2 logical cores and 1 physical core
Intel, 2 physical cores
Intel, 2 physical cores
Intel, 1 logical core and 2 physical cores
Intel, 2 logical cores and 2 physical cores
Intel, 1 CPU
Intel, 1 CPU
Intel, 1 CPU, 1 logical core
Intel, 1 CPU, 2 logical cores
Intel, 1 CPU
Intel, 1 CPU
Intel, 1 CPU, 1 logical core
Intel, 1 CPU, 2 logical cores
Intel, 1 CPU, 1 physical core
Intel, 1 CPU, 1 physical core
Intel, 1 CPU, 1 logical core and 1 physical core
Intel, 1 CPU, 2 logical cores and 1 physical core
Intel, 1 CPU, 2 physical cores
Intel, 1 CPU, 2 physical cores
Intel, 1 CPU, 1 logical core and 2 physical cores
Intel, 1 CPU, 2 logical cores and 2 physical cores
Intel, 2 CPU
Intel, 2 CPU
Intel, 2 CPU, 1 logical core
Intel, 2 CPU, 2 logical cores
Intel, 2 CPU
Intel, 2 CPU
Intel, 2 CPU, 1 logical core
Intel, 2 CPU, 2 logical cores
Intel, 2 CPU, 1 physical core
Intel, 2 CPU, 1 physical core
Intel, 2 CPU, 1 logical core and 1 physical core
Intel, 2 CPU, 2 logical cores and 1 physical core
Intel, 2 CPU, 2 physical cores
Intel, 2 CPU, 2 physical cores
Intel, 2 CPU, 1 logical core and 2 physical cores
Intel, 2 CPU, 2 logical cores and 2 physical cores
